cdap-api/src/main/java/co/cask/cdap/api/plugin/Requisites.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.plugin;
+
+import co.cask.cdap.api.annotation.Requirements;
+
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * A class to encapsulate all the different types of {@link Requirements}.
+ */
+public class Requisites {
+
+  private static final Requisites EMPTY_REQUISITES = new Requisites(Collections.emptySet());
```

Comment:
let's make this public and named EMPTY. It is a pretty established pattern to have a public static singleton object like that.